File: src/Purchase/Recommendations.Purchase.Shared/Commands/CreateOrder.cs
using Recommendations.Purchase.Shared.DTO;
using Recommendations.Shared.Abstractions.Commands.CommandWithResult;

namespace Recommendations.Purchase.Shared.Commands;

public sealed record CreateOrder(    
    Guid CustomerId,
    Guid ShippingAddressId,
    int Status,
    DateTime CreatedAt,
    DateTime? PaidAt,
    IReadOnlyCollection<OrderItemDto> Items,
    IEnumerable<PaymentDto>? Payments) : ICommandWithResult<Guid>;

File: fe/src/types/purchase/CreateOrder.ts
import type { OrderItem } from "./OrderItem";
import type { Payment } from "./Payment";

export interface CreateOrder {
  CustomerId: string;
  ShippingAddressId: string;
  Status: number;
  CreatedAt: Date;
  PaidAt: Date;
  Items: OrderItem[];
  Payments: Payment[];
}
